pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -516,14 +516,13 @@ func (a *FlavorAssigner) findFlavorForPodSetResource(
 	podSpec := &ps.Template.Spec
 
 	var bestAssignment ResourceAssignment
+	bestAssignmentIdx := -1
 	bestAssignmentMode := noFit
 
 	// We will only check against the flavors' labels for the resource.
 	selector := flavorSelector(podSpec, resourceGroup.LabelKeys)
-	attemptedFlavorIdx := -1
 	idx := a.wl.LastAssignment.NextFlavorToTryForPodSetResource(psID, resName)
 	for ; idx < len(resourceGroup.Flavors); idx++ {
-		attemptedFlavorIdx = idx
```

Comment:
Removing this line might be risky as I see it was explicitly added in https://github.com/kubernetes-sigs/kueue/pull/2407. Unfortunately back then we didn't add a 'functional' test for scheduling.

I think the proper path forward will be to:
- add a "functional" test case in schedule_test.go for the previous fix
- add a functional test case for your fix

Reply:
Opened https://github.com/kubernetes-sigs/kueue/issues/4663, I would really like to start with that.

Reply:
SGTM, thank you!